### ISSUE TITLE
Stop bumping go directive unless necessitated by other dependencies

### ISSUE THREAD

**Issue by kaovilai** [containers/image#2664]
This PR https://github.com/containers/image/pull/2601/files for example did not need to bump go directive.

The minimum go required for that PR dependency bump is 

```
❯ go mod graph | grep 'github.com/sigstore/sigstore' | grep go@
github.com/sigstore/sigstore@v1.8.11 go@1.22.0
```

A more acceptable minimum is one that is unfortunate but required by other modules.
```
❯ go mod graph | grep go@1.22.6  
github.com/sigstore/fulcio@v1.6.4 go@1.22.6
```

This repo by itself should not be enforcing minimum on other repositories importing it. Stop spreading "minimum virus"

toolchain version used will be defined outside of go.mod ideally, such as by installing a newer compatible go toolchain to ci/cd/development env.

Failing that, toolchain directive should be used instead of go directive for bumping versions to not cascade minimum versions to importing dependencies.
> [toolchain directive, in contrast to the go directive, applies only to the current module (the one defined by the go.mod file). It suggests the toolchain to be used when in that very module, and doesn't propagate to other modules.](https://github.com/golang/go/issues/70853#issuecomment-2549206026)

High profile repos that have removed/reduced minimum go patch version per user requests
- https://github.com/grpc/grpc-go/pull/7831 [motivation](https://github.com/grpc/grpc-go/pull/7624#discussion_r1757159186)

Being proactive to prevent following from reoccuring
- https://github.com/kopia/kopia/issues/4292
- https://github.com/vmware-tanzu/velero/issues/8397
- https://github.com/kubevirt/kubevirt-velero-plugin/pull/292#issuecomment-2482634796

**Comment by mtrmac** [comment]
Thanks for reaching out.

> This PR https://github.com/containers/image/pull/2601/files for example did not need to bump go directive.

That PR did need to bump the `go` directive; the PR was filed by a bot, and the bot documents the rationale.

> The minimum go required for that PR dependency bump is
> 
> ```
> ❯ go mod graph | grep 'github.com/sigstore/sigstore' | grep go@
> github.com/sigstore/sigstore@v1.8.11 go@1.22.0
> ```

That’s true as of `v1.8.11`, not as of `v1.8.10`. …`.11` decreased the required Go version.


> This repo by itself should not be enforcing minimum on other repositories importing it. Stop spreading "minimum virus"

This is an intentional property of the Go module / versioning system, and, for better or worse, no individual repo can stop that from happening. It would take a coordinated action from _way too many_ actors in the ecosystem to give users back the desired control.

I think the best bet would be to convince Go upstream to change the design, and the chances seem low to me.

Failing that, I think users need to be ready to accept the need to update to a new Go 1.X.Y version on short notice. (We do try hard to delay updates to 1.X versions, because those tend to require more migration. But that’s still a trade-off, WRT making it possible to fix vulnerabilities in dependencies quickly.)


> Failing that, toolchain directive should be used instead of go directive for bumping versions to not cascade minimum versions to importing dependencies.

When dependencies of c/image update the `go` dependency, we don’t have a choice; the tooling will enforce the minimum.

---

In this particular case _as of this moment_, we might be able to go back to 1.22.6 (I would accept a PR for that but I’m not going to file one myself) — but, I think that longer-term this is mostly futile and, sadly, users of Go-written software ultimately need to be ready to update the toolchain.

**Comment by kaovilai** [comment]
Yes the tooling will enforce the minimum for other deps of c/image. Except - some repos do actually bump go directive for forcing GitHub actions setup-go to select version based on go.mod. the proper thing to use there is toolchain directive not go directive.

Just adding cautionary tales.

There are enough coordinated effort out there that without changes to golang a significant number of projects can ward off minimum patch version of golang.

**Comment by mtrmac** [comment]
I don’t think there is anything we can specifically _change_ to make things better. E.g. right now, c/image is much behind many of the core libraries, which have already started requiring Go 1.23. (But we will follow, and move to Go 1.23, quite soon.)

**Comment by kaovilai** [comment]
What I mean here is stay on lowest compatiblity patch version. So in this case, 1.23.0 is sufficient. 

But not bump to say 1.23.6 on the go directive.
